src/util/amount.rs

```diff
@@ -0,0 +1,367 @@
+#![allow(missing_docs)]
+
+use std::error;
+use std::fmt;
+use std::ops;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Denomination {
+    Bitcoin = 8,
+    MilliBitcoin = 5,
+    Bit = 2,
+    Satoshi = 0,
+    MilliSatoshi = -3,
+}
+
+impl fmt::Display for Denomination {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(match self {
+            Denomination::Bitcoin => "BTC",
+            Denomination::MilliBitcoin => "mBTC",
+            Denomination::Bit => "bits",
+            Denomination::Satoshi => "satoshi",
+            Denomination::MilliSatoshi => "msat",
+        })
+    }
+}
+
+pub trait Precision: Default + Clone {
+    /// The number of decimal places more than a satoshi.
+    fn scale() -> i32;
+}
+
+#[derive(Default, Clone)]
+pub struct Bitcoin();
```

Comment:
This seems really useless as a Precision.

Reply:
Yeah like I said before, will be removing that one. Just have to redo the `IntoBitcoin` stuff then.